x-pack/plugins/ml/public/application/util/index_utils_provider.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { DataView, DataViewsContract } from '@kbn/data-views-plugin/public';
+import type { Job } from '../../../common/types/anomaly_detection_jobs';
+
+export function indexUtilsProvider(dataViewsService: DataViewsContract) {
```

Comment:
Suggest to rename this to `indexServiceFactory` and the file to `index_service.ts` to be in line with how we renamed the time buckets service.